src/session.c

```diff
@@ -1469,6 +1470,30 @@ libssh2_session_get_timeout(LIBSSH2_SESSION * session)
     return session->api_timeout;
 }
 
+/* libssh2_session_set_read_timeout
+ *
+ * Set a session's timeout (in sec) when reading packets,
+ * or 0 to use default of 60 seconds.
+ */
+LIBSSH2_API void
+libssh2_session_set_read_timeout(LIBSSH2_SESSION * session, long timeout)
+{
+    if(timeout == 0) {
```

Comment:
Would it be better to check for `timeout <= 0` here?

Reply:
Yes, that's probably a good plan, I'll do that and push an update.  Do you think that's the last bit for this?

Reply:
Thanks! Yes, with that I'll merge after CI is done.